core/src/main/scala/org/apache/spark/CacheManager.scala

```diff
@@ -136,7 +147,13 @@ private[spark] class CacheManager(blockManager: BlockManager) extends Logging {
        */
       blockManager.putIterator(key, values, level, tellMaster = true, effectiveStorageLevel)
       blockManager.get(key) match {
-        case Some(v) => v.data.asInstanceOf[Iterator[T]]
+        case Some(v) =>
```

Comment:
What's the consequence of forgetting to release? Does it get caught somewhere else and released as a catch all? Similar to the other comments, this makes it seem like a try {} finally{} is needed after the get(key).